File: Homework/themoviedb/src/main/java/com/example/themoviedb/persistence/MovieDao.java
package com.example.themoviedb.persistence;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Flowable;

@Dao
public interface MovieDao {

    @Query("SELECT * FROM movies WHERE title == (:title)")
    Flowable<Movie> getMovie(String title);

    @Query("SELECT * FROM movies")
    Flowable<List<Movie>> getAllMovies();

    @Query("SELECT * FROM movies WHERE iswatched == 1")
    Flowable<List<Movie>> getWatchedMovies();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insertMovie(Movie movie);

    @Delete()
    Completable deleteMovie(Movie movie);
}

File: Homework/themoviedb/src/main/java/com/example/themoviedb/persistence/MovieDao.kt
package com.example.themoviedb.persistence

import androidx.room.*
import io.reactivex.Completable
import io.reactivex.Flowable
import io.reactivex.Maybe

@Dao
interface MovieDao {

    @get:Query("SELECT * FROM movies")
    val allMovies: Flowable<List<Movie>>

    @get:Query("SELECT * FROM movies WHERE iswatched == 1")
    val watchedMovies: Flowable<List<Movie>>

    @Query("SELECT * FROM movies WHERE title == (:title)")
    fun getMovie(title: String?): Maybe<Movie>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertMovie(movie: Movie): Completable

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun updateMovie(movie: Movie): Completable

    @Delete
    fun deleteMovie(movie: Movie): Int
}
